jupyter_server/extension/application.py

```diff
@@ -411,18 +415,18 @@ def stop(self):
         self.serverapp.clear_instance()
 
     @classmethod
-    def load_jupyter_server_extension(cls, serverapp, argv=[], **kwargs):
+    def load_jupyter_server_extension(cls, serverapp):
         """Initialize and configure this extension, then add the extension's
         settings and handlers to the server's web application.
         """
         # Configure and initialize extension.
-        extension = cls()
-        extension.initialize(serverapp, argv=argv)
+        extension = cls(parent=serverapp)
+        extension.initialize(serverapp)
```

Comment:
This seems pretty significant, what is the logic behind it? I.e. why are we no longer passing `argv` to the extension?

Reply:
Argv is parsed by the ServerApp directly and the ExtensionApp traits are inherited by this ServerApp via `parent=serverapp`. Previously, we were double parsing the args—once in the ServerApp and again in the ExtensionApp.